error after recent raspi-libcamera updates

error msg:

> /usr/bin/camera-streamer Version: 0.2.8 (bc23191)
> util/http/http.c: ?: HTTP listening on 0.0.0.0:8080.
> [36:40:49.799225057] [8727]  INFO Camera camera_manager.cpp:284 libcamera v0.1.0+118-563cd78e
> [36:40:49.886218331] [8746] ERROR RPI pipeline_base.cpp:802 Failed to load a suitable IPA library
> [36:40:49.886374059] [8746] ERROR RPI vc4.cpp:216 Failed to register camera imx219 10-0010: -22
> device/libcamera/device.cc: CAMERA: No available cameras
> device/libcamera/device.cc: CAMERA: Camera `/base/soc/i2c0mux/i2c@1/imx219@10` was not found.
> device/device.c: CAMERA: Can't open device: /base/soc/i2c0mux/i2c@1/imx219@10

but camera is available!

> $ libcamera-hello --list-cameras
> Available cameras
> 0 : imx219 [3280x2464 10-bit RGGB] (/base/soc/i2c0mux/i2c@1/imx219@10)

compiling it manually also fails...

> device/libcamera/options.cc: In function ‘int libcamera_device_dump_control_option(int (*)(device_option_t*, void*), void*, const libcamera::ControlId&, const libcamera::ControlInfo*, const libcamera::ControlValue*, bool)’:
> device/libcamera/options.cc:183:10: error: enumeration value ‘ControlTypePoint’ not handled in switch [-Werror=switch]
>   183 |   switch (control_id.type()) {
>       |          ^
> device/libcamera/options.cc: In function ‘int libcamera_device_set_option(device_t*, const char*, const char*)’:
> device/libcamera/options.cc:400:12: error: enumeration value ‘ControlTypePoint’ not handled in switch [-Werror=switch]
>   400 |     switch (control_id->type()) {
>       |            ^
> ...
> make[1]: Leaving directory '/home/pi/camera-streamer'
> make: *** [Makefile:67: all] Error 2

recent raspi-libcamera updates:

> Preparing to unpack .../01-libcamera-ipa_0.3.2+rpt20241119-1_armhf.deb ...
> Unpacking libcamera-ipa:armhf (0.3.2+rpt20241119-1) over (0.3.2+rpt20240927-1) ...
> Preparing to unpack .../02-libcamera0.3_0.3.2+rpt20241119-1_armhf.deb ...
> Unpacking libcamera0.3:armhf (0.3.2+rpt20241119-1) over (0.3.2+rpt20240927-1) ...
> Preparing to unpack .../03-rpicam-apps_1.5.3-1_armhf.deb ...
> Unpacking rpicam-apps (1.5.3-1) over (1.5.2-2) ...
> Preparing to unpack .../04-libcamera-apps_1.5.3-1_all.deb ...
> Unpacking libcamera-apps (1.5.3-1) over (1.5.2-2) ...
> Preparing to unpack .../05-linux-libc-dev_1%3a6.6.62-1+rpt1_all.deb ...
> Unpacking linux-libc-dev (1:6.6.62-1+rpt1) over (1:6.6.51-1+rpt3) ...
> Preparing to unpack .../06-python3-libcamera_0.3.2+rpt20241119-1_armhf.deb ...
> Unpacking python3-libcamera:armhf (0.3.2+rpt20241119-1) over (0.3.2+rpt20240927-1) ...
> Preparing to unpack .../07-python3-picamera2_0.3.23-1_all.deb ...
> Unpacking python3-picamera2 (0.3.23-1) over (0.3.22-2) ...

--- a/device/libcamera/options.cc
+++ b/device/libcamera/options.cc
@@ -359,6 +359,7 @@ static libcamera::Size libcamera_parse_size(const char *value)
   return libcamera::Size();
 }
 
+#if LIBCAMERA_VERSION_MAJOR == 0 && LIBCAMERA_VERSION_MINOR > 3 && LIBCAMERA_VERSION_PATCH >= 2 // Support for older libcamera versions
 static libcamera::Point libcamera_parse_point(const char *value)
 {
   static const char *POINT_PATTERNS[] =
@@ -379,6 +380,7 @@ static libcamera::Point libcamera_parse_point(const char *value)
 
   return libcamera::Point();
 }
+#endif
 
 template<typename T, typename F>
 static bool libcamera_parse_control_value(libcamera::ControlValue &control_value, const char *value, const F &fn)

